pkg/api/v1/types.go

```diff
@@ -2030,6 +2030,9 @@ type SecurityContext struct {
 
 	// RunAsUser is the UID to run the entrypoint of the container process.
 	RunAsUser *int64 `json:"runAsUser,omitempty" description:"the user id that runs the first process in the container; see http://releases.k8s.io/HEAD/docs/design/security_context.md#security-context"`
+
+	// Run the container using the host ipc mode
+	UseHostIpc *bool `json:"useHostIpc,omitempty" description:"run the container using the host ipc mode; see http://releases.k8s.io/HEAD/docs/design/security_context.md#security-context"`
```

Comment:
On Aug 11, 2015, at 6:19 PM, Vish Kannan <notifications@github.com> wrote:

In pkg/api/v1/types.go
<https://github.com/GoogleCloudPlatform/kubernetes/pull/12470#discussion_r36806530>
:

> @@ -2030,6 +2030,9 @@ type SecurityContext struct {
>
>  	// RunAsUser is the UID to run the entrypoint of the container process.
>  	RunAsUser *int64 `json:"runAsUser,omitempty" description:"the user id that runs the first process in the container; see http://releases.k8s.io/HEAD/docs/design/security_context.md#security-context"`
> +
> +	// Run the container using the host ipc mode
> +	UseHostIpc *bool `json:"useHostIpc,omitempty" description:"run the container using the host ipc mode; see http://releases.k8s.io/HEAD/docs/design/security_context.md#security-context"`

+1 to moving namespace sharing to Pod level. It gets confusing once we
start supporting namespace sharing among containers in a pod.
Instead of sharing namespaces with host piecemeal, can we have one knob
that results in the pod sharing most of the namespaces with the host? I
assume this kind of a pod is already privileged and we trust it.


Tim proposed moving security context to the pod scope in the volume
security context issue (@pmorie / @thockin I don't remember the number).
However, I raised the concern that we have use cases where a few attributes
should still remain with the container, namely user and group ids (and
possibly whether the user is namespaces).  We have a fairly large number of
customer containers that are dependent on matching host uids, and having
one uid per pod makes that impossible.

My proposal would be that we move security context to the pod scope,
including other namespace settings.  However, I would like to preserve at
minimum per container uid, gids, and maybe selinux labels (and others that
overlap) on the container, where they can override the pod in specific
cases.  This would allow a v1 compatible change.  We could then apply pod
security context to volumes (which gives Tim what he wanted there, namely
that you can't have disjoint volumes mounting rules).

@pweil-

If we did that, security context is probably the right place to leave this.


—
Reply to this email directly or view it on GitHub
<https://github.com/GoogleCloudPlatform/kubernetes/pull/12470/files#r36806530>
.